formats/AtomFormat.php

```diff
@@ -27,11 +36,33 @@ public function stringify(){
 
 		$entries = '';
 		foreach($this->getItems() as $item) {
-			$entryAuthor = $this->xml_encode($item->getAuthor());
-			$entryTitle = $this->xml_encode($item->getTitle());
-			$entryUri = $this->xml_encode($item->getURI());
-			$entryTimestamp = $this->xml_encode(date(DATE_ATOM, $item->getTimestamp()));
-			$entryContent = $this->xml_encode($this->sanitizeHtml($item->getContent()));
+			$entryTimestamp	= $item->getTimestamp();
+			$entryTitle		= $this->xml_encode($item->getTitle());
+			$entryContent	= $item->getContent();
+			$entryUri		= $item->getURI();
+
+			// the item id must be a valid unique URI
+			$entryID = $this->xml_encode($entryUri);
+			if (empty($entryID))
+				$entryID = 'urn:sha1:' . hash('sha1', $entryTitle . $entryContent);
+
+			if (empty($entryTimestamp))
+				$entryTimestamp = $this->lastModified;
```

Comment:
Leave it empty if it's not specified. The specification does not require a time stamp for each item in the feed.

Reply:
According to RFC 4287 section 4.1.2, It is mandatory:
> atom:entry elements MUST contain exactly one atom:updated element.

Reply:
You are right. I'm positive I checked before commenting, but obviously I'm unable to read.